vmware-base/src/main/java/com/cloud/hypervisor/vmware/mo/DatacenterMO.java

```diff
@@ -338,10 +341,22 @@ public List<ObjectContent> getVmPropertiesOnDatacenterVmFolder(String[] property
         PropertyFilterSpec pfSpec = new PropertyFilterSpec();
         pfSpec.getPropSet().add(pSpec);
         pfSpec.getObjectSet().add(oSpec);
-        List<PropertyFilterSpec> pfSpecArr = new ArrayList<PropertyFilterSpec>();
+        List<PropertyFilterSpec> pfSpecArr = new ArrayList<>();
         pfSpecArr.add(pfSpec);
 
-        return _context.getService().retrieveProperties(_context.getPropertyCollector(), pfSpecArr);
+        RetrieveOptions ro = new RetrieveOptions();
+        if (maxObjects != null && maxObjects > 0) {
+            ro.setMaxObjects(maxObjects);
+        }
+
+        RetrieveResult result = _context.getService().retrievePropertiesEx(_context.getPropertyCollector(), pfSpecArr, ro);
+        return createReturnObjectPair(result);
+    }
+
+    private static Pair<String, List<ObjectContent>> createReturnObjectPair(RetrieveResult result) {
+        String tokenForRetrievingNewResults = result.getToken();
```

Comment:
Nice solution, can we also add to the javadocs the meaning of this token regarding Vmware SDK? (maybe also a link to the SDK docs): `the token returned in the previous RetrieveResult returned on the same session by the same PropertyCollector.` (https://vdc-download.vmware.com/vmwb-repository/dcr-public/4fcedce4-0e06-49cc-bb39-38732c40995d/c087e649-192b-416c-9aa1-dcbb75e74d17/vmodl.query.PropertyCollector.html - maybe this is not the latest URL)